pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -362,6 +587,9 @@ func (c *Controller) constructWorkload(pod *v1.Pod, waypoints []*v1.Pod) *worklo
 	if !IsPodReady(pod) {
 		return nil
 	}
+	if pod.Spec.HostNetwork {
```

Comment:
for debug-ability, maybe we can store an "ignored" index with minimal info or emit a debug-level log when we ignore something. implicit rules like this are often the source of users' ?s